.travis.yml

```diff
@@ -17,11 +17,12 @@ language: python
 python:
   - '2.7'
 env:
-  - PYARROW_VERSION=0.12.1 TF_VERSION=1.12 DEPLOYABLE=0 PY=2.7 PYSPARK_VERSION=2.4.4
-  - ARROW_PRE_0_15_IPC_FORMAT=1 PYARROW_VERSION=0.15.1 TF_VERSION=1.12 DEPLOYABLE=0 PY=2.7 PYSPARK_VERSION=2.4.4
-  - PYARROW_VERSION=0.12.1 TF_VERSION=1.13.2 DEPLOYABLE=0 PY=3.7 PYSPARK_VERSION=2.4.4
-  - ARROW_PRE_0_15_IPC_FORMAT=1 PYARROW_VERSION=0.15.1 TF_VERSION=1.13.2 DEPLOYABLE=1 PY=3.7 PYSPARK_VERSION=2.4.4
-  - ARROW_PRE_0_15_IPC_FORMAT=0 PYARROW_VERSION=0.15.1 TF_VERSION=1.15.0 DEPLOYABLE=1 PY=3.7 PYSPARK_VERSION=3.0.0
+  - PYARROW_VERSION=0.12.1 TF_VERSION=1.15.0 DEPLOYABLE=0 PY=2.7 PYSPARK_VERSION=2.4.4
+  - PYARROW_VERSION=0.15.1 TF_VERSION=1.15.0 DEPLOYABLE=0 PY=2.7 PYSPARK_VERSION=2.4.4 ARROW_PRE_0_15_IPC_FORMAT=1
+  - PYARROW_VERSION=0.12.1 TF_VERSION=1.15.0 DEPLOYABLE=0 PY=3.7 PYSPARK_VERSION=2.4.4
+  - PYARROW_VERSION=0.15.1 TF_VERSION=1.15.0 DEPLOYABLE=1 PY=3.7 PYSPARK_VERSION=2.4.4 ARROW_PRE_0_15_IPC_FORMAT=1
```

Comment:
@selitvin 
See https://spark.apache.org/docs/3.0.0-preview/sql-pyspark-pandas-with-arrow.html#compatibiliy-setting-for-pyarrow--0150-and-spark-23x-24x

Summary:
For spark 2.3/2.4 with pyarrow<0.15, no ARROW_PRE_0_15_IPC_FORMAT setting and works well.
For spark 2.3/2.4 with pyarrow 0.15+, set ARROW_PRE_0_15_IPC_FORMAT=1, otherwise may raise error when spark read parquet array fields.
For spark 3.0 with pyarrow 0.15+, set ARROW_PRE_0_15_IPC_FORMAT=0 (default behavior)

And, which one should be deployable ?

Reply:
It does not matter much which one we choose as deployable. I'd choose (arbitrary) the one with the most up to date dependencies.